### ISSUE TITLE
[Documentation] Provide a baseline infrastructure description for development

### ISSUE THREAD

**Issue by cwatsonc** [freeboardgames/FreeBoardGames.org#989]
Description: The fbg documentation doesn't currently prescribe minimum infra required to develop/build the solution.  The designers have chosen monorepo and choices for improvements are driven by currently supported platform and versions.

Suggestion: Provide a baseline support document describing our platform so developers may align their efforts with the community and ease adoption of our framework.

Proposed:
Provide a platform minimal description that is supported in the current build.  Annotate the package.json with properties that which will warn/fail process if not present.  If yarn is our chosen build tool, create a minimum version which is supported to give guidance on tools choice. Add an "engines" property in package.json to enforce/warn minimums at build time. Add "cpu" and "os" descriptions with similar prescriptive advice. Make these mutable to allow speculative development but matching should guarantee a successful build and easy integration of contributions by development. 

Alternatives Considered:
I have been reading the yarn docs and looking forward to the yarn 2.x.x upgrades -- I have been reading multiple articles on alternatives for build orchestration (e.g. bit, npm).  Yarn 2 is a radical shift in build behavior and structure but promises swift operational builds on first download.  I will compile a list of articles discussing the rationales.

Additional Context:
[yarn 2 migration and rationales](https://yarnpkg.com/getting-started/migration)
[Alternative Package Managers](https://blog.bitsrc.io/4-npm-alternatives-best-js-package-managers-and-publishing-tools-fe6779937ee9)
[Yarn manifest docs](https://yarnpkg.com/configuration/manifest)
[Engines Plugin for Yarn](https://github.com/devoto13/yarn-plugin-engines)

**Comment by cwatsonc** [comment]
created PR #1013

**Comment by richardaum** [comment]
> Yarn 2 is a radical shift in build behavior and structure

Yes, it is, definetely. And I think in that case the costs overcome the benefits. I mean, Yarn 1 is doing a good job here, right?

**Comment by cwatsonc** [comment]
@richardaum the issue is discussed on the "rationales" link on the main issue -- yarn@1.0x.x is about to be orphaned for support -- the yarn team is poised to update to 3.x.x in the next 6 months and will ensure that fact.  

The risk of staying on orphaned versions outweighs any risk of moving forward.  @ProspectPyxis agrees with that in comments but was thinking moving to an alternative pnpm -- that has a bigger risk of than an upgrade of our existing tooling. As is mentioned in the rationale -- many of the improvements are initially OPTIONAL and don't require immediate adoption -- it's a much more graceful jump than ripping and replacing with an untested alternative IMO

See Also [yarn 2 migration and rationales](https://yarnpkg.com/getting-started/migration)
See Also [alternative package managers](https://blog.bitsrc.io/4-npm-alternatives-best-js-package-managers-and-publishing-tools-fe6779937ee9)
